src/s1reader/s1_burst_slc.py

```diff
@@ -579,6 +585,332 @@ def az_carrier_components(self, offset, position):
+
+        grid_freq_dcg = self._evaluate_polynomial_array(arr_coeff_a,
+                                                        grid_tau,
+                                                        tau0_fdc_interp)
+
+        grid_freq_dc = (grid_freq_dcg
+                        + grid_kappa_t * ((grid_t - t_burst)
+                                          + (freq_dcg_burst / kappa_annotation_burst
+                                          - grid_freq_dcg / kappa_annotation_grid)))
+
+        with (gdal.Open(list_filename_llh[0], gdal.GA_ReadOnly) as raster_lat,
+              gdal.Open(list_filename_llh[1], gdal.GA_ReadOnly) as raster_lon,
+              gdal.Open(list_filename_llh[2], gdal.GA_ReadOnly) as raster_hgt):
+            lat_map = raster_lat.ReadAsArray()
+            lon_map = raster_lon.ReadAsArray()
+            hgt_map = raster_hgt.ReadAsArray()
+
+        x_ecef, y_ecef, z_ecef = self._llh_to_ecef(lat_map,
+                                                   lon_map,
+                                                   hgt_map,
+                                                   ellipsoid)
+
+        # Populate the position, velocity, and
+        # acceleration of the satellite to the correction grid
+        list_pos_satellite = []
+        list_vel_satellite = []
+        list_acc_satellite = []
+
+        for i in range(3):
+            list_pos_satellite.append(vec_position_intp[:, i][..., np.newaxis]
+                                      * np.ones(grid_tau.shape[1])[np.newaxis, ...])
+            list_vel_satellite.append(vec_vel_intp[:, i][..., np.newaxis]
+                                      * np.ones(grid_tau.shape[1])[np.newaxis, ...])
+            list_acc_satellite.append(vec_acceleration_intp[:, i][..., np.newaxis]
+                                      * np.ones(grid_tau.shape[1])[np.newaxis, ...])
+
+        x_s, y_s, z_s = list_pos_satellite
+        vx_s, vy_s, vz_s = list_vel_satellite
+        ax_s, ay_s, az_s = list_acc_satellite
+        
+        mag_xs_xg = np.sqrt(  (x_s - x_ecef)**2
+                            + (y_s - y_ecef)**2
+                            + (z_s - z_ecef)**2)
+
+        dotp_dxsg_acc = (  (x_s - x_ecef) * ax_s
+                         + (y_s - y_ecef) * ay_s
+                         + (z_s - z_ecef) * az_s)
+
+        kappa_annotation_true = ( -(2 * self.radar_center_frequency)
+                                 / (isce3.core.speed_of_light * mag_xs_xg)
+                                 * (dotp_dxsg_acc + (vx_s**2 + vy_s**2 + vz_s**2)))
+
+        delta_t_freq_mm = (grid_freq_dc
+                           * (-1 / kappa_annotation_grid
+                              + 1 / kappa_annotation_true))
+
+        #return delta_t_freq_mm
+
+        # Prepare to export to LUT2d
+        vec_range = grid_tau(index_mid_burst_t) * isce3.speed_of_light / 2.0
+
+        return isce3.core.LUT2d(vec_range, vec_t, delta_t_freq_mm)
```

Comment:
If I am not mistaken, for COMPASS, we have returned a LUT where the the first two vectors would define the grid on which the correction is computed in range and azimuth. We chose to express the vector in range using `meters` and the one in azimuth using `seconds`.

Reply:
I think we are on the same understandings. The computation in the function `az_fm_rate_mismatch_mitigation()` is based on the grid whose unit in azimuth and range are seconds (by following the ETAD ATBD).
The line 910 above is to convert the range grid in time (tau) into meters, so that the return LUT2d is in compliance with our convention.